IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Functions/IfcAlignmentRepresentation/Expression.txt

```diff
@@ -0,0 +1,40 @@
+(Alignment : IfcAlignment) : IfcBoolean;
+	
+	LOCAL
+		AlignmentRepresentations : SET OF IfcShapeRepresentation := SELF.Representation.Representations;
+	END_LOCAL;
+
+	/* no representation is okay */
+	IF NOT EXISTS(SELF.Representation)
+		RETURN TRUE;
+	END_IF;
+	
+	/* one representation is either Curve2D or 3D */
+	IF SIZEOF(AlignmentRepresentations) = 1
+	AND
+	(SIZEOF(QUERY(twoDee <* AlignmentRepresentations | twoDee.RepresentationType = 'Curve2D')) = 1
+	OR
+	SIZEOF(QUERY(threeDee <* AlignmentRepresentations | threeDee.RepresentationType = 'Curve3D')) = 1)
+		RETURN TRUE;
+	END_IF;
+	
+	/* 2 representations and 3D BaseCurve reference equals 2D representation */
+	IF SIZEOF(AlignmentRepresentations) = 2)
+	
+		TwoDRep : SET OF IfcRepresentationItem := QUERY(twoDee <* AlignmentRepresentations | twoDee.RepresentationType = 'Curve2D');
+		ThreeDRep : SET OF IfcRepresentationItem := QUERY(threeDee <* AlignmentRepresentations | threeDee.RepresentationType = 'Curve3D');
+		IF SIZEOF(ThreeDRep) = 1 AND ('IFCGEOMETRYRESOURCE.IFCGRADIENTCURVE' IN TYPEOF(ThreeDRep[1].Items[1]))
+			IF SIZEOF(TwoDRep) = 1 AND ('IFCGEOMETRYRESOURCE.IFCCOMPOSITECURVE' IN TYPEOF(TwoDRep[1].Items[1]))
```

Comment:
Should we query instead of `[1]`, in case there are multiple `Items`?

Reply:
Yes and no. A query also returns a SET of 1.